src/Parallel/Info.hpp

```diff
@@ -0,0 +1,101 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/// \file
+/// Defines functions that provide low-level system information such as number
+/// of nodes and processors, in a way that they are mockable in ActionTesting.
+
+#pragma once
+
+// The functions in namespace `sys` that return information on nodes
+// and cores are templated on DistribObject.  Actions should use
+// these functions rather than the raw charm++ versions (in the
+// sys::charm namespace below) so that the mocking framework will see
+// the mocked cores and nodes.
+namespace sys {
```

Comment:
Could you make this the `Parallel` namespace and the one in `Utilities/System` just `sys` (instead of `sys::charm`)? Then if I understand correctly this PR keeps the low-level `sys` functions the same but introduces new `Parallel` functions that work with distributed objects.

Reply:
I can put these into the `Parallel` namespace.  But I kind of like `sys::charm` for the other ones to emphasize that these are charm-dependent functions, and also to discourage people from using them (or make people think twice before using them).

One worry I have:  Suppose someone writes a new Action (or modifies an existing Action) and uses the charm functions. There will be no compiler error; the only effect will be that during ActionTesting the function will return the wrong answer (i.e. always one core and one node) and result in a bug that will be hard to find.  Thoughts?

Reply:
- Yea we can consider moving the "raw" functions from `sys` -> `sys::charm`, though I think that doesn't need to happen in this PR. I would prefer to keep them in `sys` because they are not inherently charm-specific (only their implementation calls into charm).
- I can't think of a foolproof way to make sure people use the `Parallel::` functions, other than adding warnings to the docs. The `sys` functions have legitimate use-cases in layers of code that don't yet depend on `Parallel` (mostly error-handling) to avoid cyclic dependencies.
- The interface you suggested before, where the distributed-object functions are in `sys`, has the problem that distributed objects are a `Parallel` concept that shouldn't be lifted to `sys`. Since the functions are generic the cyclic dependency is technically broken (I think). However, the only argument I can see for putting the distributed-object functions in `sys` is to "hide" the raw functions a bit more by moving the `Parallel` functions closer to them, which doesn't really address the worry that people can use the raw functions.

So my suggestion is: Keep the raw functions in `sys`. Add functions to `Parallel` that work with distributed objects. Add warnings to the `sys` functions that they are super low-level and people should use the `Parallel` functions when they work with distributed objects. What do you think?

Reply:
Ok, done!  Fixup commits pushed (and rebase onto current develop)